dcp-0003/dcp-0003.mediawiki

```diff
@@ -0,0 +1,837 @@
+their respective individual amounts they used to fund the contract, however,
+they do not broadcast them to the network.  Instead, as they conduct business,
+they create new commitment transactions which alter how the funds are split
+between them.  Since the anchor transaction can only be spent once, and due to
+the restrictions enforced by the underlying smart contract, only the most recent
+commitment transaction is valid.  Consequently, the payment channel can be
+trustlessly closed at any time by either party by simply broadcasting the
+most recent commitment transaction to the network for inclusion in the
+blockchain.
+
+Since the commitment transactions are time locked, using absolute times would
+effectively place a lifetime on the channel.  This is the case because it
+would be necessary to close and reopen the channel any time the absolute timeout
+was approaching in order to prevent the counterparty from being able to steal
+funds.  This also implies that it would be necessary to use fairly long timeouts
+with absolute time locks in order to limit the frequency of channel churn.  Long
+timeouts are not desirable because it increases the time it takes to recover
+funds from the channel in the event of an uncooperative counterparty.
+
+These problems can all be avoided by using relative time locks.
+
+===Lightning Network===
+
+The Lightning Network further builds on the concept of bidirectional payment
+channels to create a network of them over which payments can be trustlessly
+routed by finding a path similar to the way packets are routed through the
+internet.  In order to do this, it essentially makes use of HTLCs with
+decrementing time locks.
+
+As mentioned in the section about HTLCs, the use of relative time locks enables
+off-chain negotiation without requiring absolute deadlines.  Thus, relative time
+locks are important for efficient operation of LN as well as timely revocation.
+
+===Off-chain Atomic Swaps===
+
+Combining the concepts of bidirectional payment channels, routing over multiple
+hops, and techniques to perform cross-chain atomic swaps, such as those
+described in [[https://github.com/decred/dcps/blob/master/dcp-0002/dcp-0002.mediawiki|DCP0002]],
+it is possible to instantly and trustlessly exchange funds between chains with
+heterogeneous consensus rules so long as both chains support a shared
+cryptographic hash function, such as SHA-256.
+
+In short, off-chain atomic swaps provide the opportunity to create instant,
+trustless, decentralized cryptocurrency exchanges.
+
+===Escrow with Timeout===
+
+A common technique when making payments for products or services via an escrow
+is to create a 2-of-3 multsig transaction between the buyer, seller, and a
+third-party escrow agent.  In this type of setup, the escrow agent acts as an
+arbiter in the event the transaction does not go smoothly, but is not able to
+unilaterally steal the funds.
```

Comment:
Not sure if references here are appropriate, but perhaps to BitSquare.

Reply:
I generally prefer not to link to commercial services from technical proposals.  Typically the references are limited to other technical documentation and whitepapers.  In this particular case, the escrow transactions are common knowledge that I didn't need to use any reference for, hence I didn't include anything.